src/main/java/bio/terra/app/controller/DatasetsApiController.java

```diff
@@ -248,6 +250,20 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
     return jobToResponse(jobService.retrieveJob(jobId, userReq));
   }
 
+  @Override
+  public ResponseEntity<BulkLoadHistoryModelList> getLoadHistoryForLoadTag(
+      @PathVariable("id") UUID id,
+      @PathVariable("loadTag") String loadTag,
+      @Valid @RequestParam(value = "offset", required = false, defaultValue = "0") Integer offset,
+      @Valid @RequestParam(value = "limit", required = false, defaultValue = "10") Integer limit) {
+    iamService.verifyAuthorization(
+        getAuthenticatedInfo(), IamResourceType.DATASET, id.toString(), IamAction.READ_DATASET);
+    List<BulkLoadHistoryModel> history = datasetService.getLoadHistory(id, loadTag, offset, limit);
+
+    var responseModel = new BulkLoadHistoryModelList().total(history.size()).items(history);
+    return new ResponseEntity<>(responseModel, HttpStatus.OK);
```

Comment:
It's totally a minor thing but I think the convenience methods on `ResponseEntity` are easier to read and write.
```suggestion
    return ResponseEntity.ok(new BulkLoadHistoryModelList().total(history.size()).items(history));
```